docs/src/docs/asciidoc/core-model-components.adoc

```diff
@@ -0,0 +1,470 @@
+
+The provided implementations of `OAuth2AuthorizationService` are `InMemoryOAuth2AuthorizationService` and `JdbcOAuth2AuthorizationService`.
+The `InMemoryOAuth2AuthorizationService` implementation stores `OAuth2Authorization` instances in-memory and is recommended *ONLY* to be used during development and testing.
+`JdbcOAuth2AuthorizationService` is a JDBC implementation that persists `OAuth2Authorization` instances by using `JdbcOperations`.
+
+[NOTE]
+The `OAuth2AuthorizationService` is an *OPTIONAL* component and defaults to `InMemoryOAuth2AuthorizationService`.
+
+The following example shows how to register an `OAuth2AuthorizationService` `@Bean`:
+
+[source,java]
+----
+@Bean
+public OAuth2AuthorizationService authorizationService() {
+	return new InMemoryOAuth2AuthorizationService();
+}
+----
+
+Alternatively, configure the `OAuth2AuthorizationService` through the xref:configuration-model.adoc#oauth2-authorization-server-configurer[`OAuth2AuthorizationServerConfigurer`]:
+
+[source,java]
+----
+@Bean
+public SecurityFilterChain authorizationServerSecurityFilterChain(HttpSecurity http) throws Exception {
+	OAuth2AuthorizationServerConfigurer<HttpSecurity> authorizationServerConfigurer =
+		new OAuth2AuthorizationServerConfigurer<>();
+	http.apply(authorizationServerConfigurer);
+
+	authorizationServerConfigurer
+		.authorizationService(authorizationService);
+
+	...
+
+	return http.build();
+}
+----
+
+[[oauth2-authorization-consent]]
+== OAuth2AuthorizationConsent
+
+An `OAuth2AuthorizationConsent` is a representation of an authorization "consent" (decision) from an https://datatracker.ietf.org/doc/html/rfc6749#section-4.1.1[OAuth2 Authorization Request flow] – for example, the `authorization_code` grant, which holds the authorities granted to a <<registered-client, client>> by the resource owner.
+
+When authorizing access to a client, the resource owner may grant only a subset of the authorities requested by the client.
+The typical use case is the `authorization_code` grant flow, in which the client requests scope(s) and the resource owner grants (or denies) access to the requested scope(s).
+
+After the completion of an OAuth2 Authorization Request flow, an `OAuth2AuthorizationConsent` is created (or updated) and associates the granted authorities with the client and resource owner.
+
+`OAuth2AuthorizationConsent` and its attributes are defined as follows:
+
+[source,java]
+----
+public final class OAuth2AuthorizationConsent implements Serializable {
+	private final String registeredClientId;    <1>
+	private final String principalName; <2>
+	private final Set<GrantedAuthority> authorities;    <3>
+
+	...
+
+}
+----
+<1> `registeredClientId`: The ID that uniquely identifies the <<registered-client, RegisteredClient>>.
+<2> `principalName`: The principal name of the resource owner.
+<3> `authorities`: The authorities granted to the client by the resource owner. An authority can represent a scope, claim, permission, role, etc.
```

Comment:
Change "a scope, claim, permission, role, etc." to "a scope, a claim, a permission, a role, and others."

Too many people (according to the studies I've read, about half) misinterpret "etc."